### DISCUSSION TITLE
How can I access the selection so that I can do some calculation (length, area) on that selection?

### DISCUSSION THREAD

**Discussion by adlan2001** [DotSpatial/DotSpatial#1518] [Q&A]
As the title has stated, I'm facing this problem. the solution is also nowhere to be found in the internet. Btw, here is my code for the button that will trigger a function to calculate the length of the selected features (the button doesn't do anything):
` private void BtnMeasureLength_Click(object sender, EventArgs e)
        {
            // Get the selected features
            IMapFeatureLayer? selectedLayerr = map1.Layers.SelectedLayer as IMapFeatureLayer;
            IFeatureSet? selectedFeatures = selectedLayerr?.Selection.ToFeatureSet();

            if (selectedFeatures != null)
            {
                // Check the feature type
                if (selectedFeatures.FeatureType == FeatureType.Point)
                {
                    MessageBox.Show("Error: Point feature type does not have a length.");
                }
                else // if (selectedFeatures.FeatureType == FeatureType.Line)
                {
                    double totalLength = 0;

                    // Calculate the length/perimeter of each line/polygon feature
                    foreach (IFeature feature in selectedFeatures.Features)
                    {
                        totalLength += feature.Geometry.Length;
                    }
                    if (selectedFeatures.FeatureType == FeatureType.Line)
                    {
                        MessageBox.Show($"Total length: {totalLength}");
                    }
                    else
                    {
                        MessageBox.Show($"Total perimeter: {totalLength}");
                    }
                }
            }`

**Comment by sindizzy** [comment]
Add a breakpoint and step through your lines of code.
